Library/Homebrew/extend/os/mac/requirements/java_requirement.rb

```diff
@@ -1,13 +1,34 @@
 class JavaRequirement < Requirement
-  cask "java"
+  def initialize(tags)
```

Comment:
Is this custom initialize actually necessary?

The initialize for `requirements/java_requirement` is

```
  def initialize(tags)
    @version = tags.shift if /(\d+\.)+\d/ =~ tags.first
    super
  end
```
and for requirement is
```
  def initialize(tags = [])
    @default_formula = self.class.default_formula
    @cask ||= self.class.cask
    @download ||= self.class.download
    @formula = nil
    tags.each do |tag|
      next unless tag.is_a? Hash
      @cask ||= tag[:cask]
      @download ||= tag[:download]
    end
    @tags = tags
    @tags << :build if self.class.build
    @name ||= infer_name
  end
```
so I think this is redundant unless I've missed some detail.